bin/behat-prepare.sh

```diff
@@ -169,78 +145,59 @@ foreach (\$config['example-userpass'] as \$key => &\$user) {
+openssl req -newkey rsa:2048 -new -x509 -days 3652 -nodes \
+  -out "$PREPARE_DIR/private/simplesamlphp/cert/saml.crt" \
+  -keyout "$PREPARE_DIR/private/simplesamlphp/cert/saml.pem" -batch
 
 TWIG_TEMPLATE_PATH="$PREPARE_DIR/private/simplesamlphp/modules/core/templates/loginuserpass.twig"
-# Modify the login template so Behat can submit the form
 echo "Operating on: $TWIG_TEMPLATE_PATH"
-sed -i  -- "s/<button class=\"pure-button pure-button-red pure-input-1-2 pure-input-sm-1-1 right\" id=\"submit_button\"/<button class=\"pure-button pure-button-red pure-input-1-2 pure-input-sm-1-1 right\" id=\"submit\"/g" "$TWIG_TEMPLATE_PATH"
-sed -i  -- "s/Login/Submit/g" "$TWIG_TEMPLATE_PATH"
-# Modify the loginuserpass.js file so Behat can submit the form
-JS_FILE_PATH="$PREPARE_DIR/private/simplesamlphp/modules/core/public/assets/js/loginuserpass.js" # Adjust path if necessary
-sed -i -- 's/getElementById("submit_button")/getElementById("submit")/g' "$JS_FILE_PATH"
-sed -i -- 's/button.disabled = true;//g' "$JS_FILE_PATH"
+sed -i -- 's/id="submit_button"/id="submit"/g' "$TWIG_TEMPLATE_PATH" || true
+sed -i -- 's/>Login</>Submit</g' "$TWIG_TEMPLATE_PATH" || true
 
-composer install --no-dev --working-dir="$PREPARE_DIR"/private/simplesamlphp --ignore-platform-req=ext-ldap
+JS_FILE_PATH="$PREPARE_DIR/private/simplesamlphp/modules/core/public/assets/js/loginuserpass.js"
+sed -i -- 's/getElementById("submit_button")/getElementById("submit")/g' "$JS_FILE_PATH" || true
+sed -i -- 's/button.disabled = true;//g' "$JS_FILE_PATH" || true
+
+composer install --no-dev --working-dir="$PREPARE_DIR/private/simplesamlphp" --ignore-platform-req=ext-ldap || true
 
-# Copy SimpleSAMLphp installation into public /simplesaml directory.
 cd "$PREPARE_DIR"
-mkdir -p "$PREPARE_DIR"/simplesaml
-cp -r "$PREPARE_DIR"/private/simplesamlphp/public/* "$PREPARE_DIR"/simplesaml
-cp -r "$PREPARE_DIR"/private/simplesamlphp/vendor "$PREPARE_DIR"/simplesaml/
-cp -r "$PREPARE_DIR"/private/simplesamlphp/src "$PREPARE_DIR"/simplesaml/
-cp -r "$PREPARE_DIR"/private/simplesamlphp/modules "$PREPARE_DIR"/simplesaml/
-cp -r "$PREPARE_DIR"/private/simplesamlphp/config "$PREPARE_DIR"/simplesaml/
-cp -r "$PREPARE_DIR"/private/simplesamlphp/templates "$PREPARE_DIR"/simplesaml/
-cp -r "$PREPARE_DIR"/private/simplesamlphp/cert "$PREPARE_DIR"/simplesaml/
-cp -r "$PREPARE_DIR"/private/simplesamlphp/metadata "$PREPARE_DIR"/simplesaml/
-cp -r "$PREPARE_DIR"/private/simplesamlphp/routing "$PREPARE_DIR"/simplesaml/
-cp -r "$PREPARE_DIR"/private/simplesamlphp/attributemap "$PREPARE_DIR"/simplesaml/
-cp -r "$PREPARE_DIR"/private/simplesamlphp/lib "$PREPARE_DIR"/simplesaml/
-
-# Modify the include...
+mkdir -p "$PREPARE_DIR/simplesaml"
+cp -r "$PREPARE_DIR/private/simplesamlphp/public/"*   "$PREPARE_DIR/simplesaml/"
+cp -r "$PREPARE_DIR/private/simplesamlphp/"{vendor,src,modules,config,templates,cert,metadata,routing,attributemap,lib} "$PREPARE_DIR/simplesaml/"
```

Comment:
While it looks like nothing here actually changes, breaking each cp to separate lines ensures that we can clearly see (and modify in the future, if necessary) each directory that's being copied. I disagree that this change is an improvement.